presto-hive/src/test/java/com/facebook/presto/hive/TestHiveLogicalPlanner.java

```diff
@@ -1562,6 +1562,38 @@ public void testMaterializedViewForUnionAll()
         }
     }
 
+    @Test
+    public void testMaterializedViewForExcept()
+    {
+        QueryRunner queryRunner = getQueryRunner();
+        String table = "test_customer_union";
+        String view = "test_customer_view_union";
+        try {
+            computeActual(format("CREATE TABLE %s WITH (partitioned_by = ARRAY['nationkey']) " +
+                    "AS SELECT custkey, name, address, nationkey FROM customer", table));
+
+            String baseQuery = format(
+                    "SELECT name, custkey, nationkey FROM %s WHERE custkey < 1000 EXCEPT " +
+                            "SELECT name, custkey, nationkey FROM %s WHERE custkey > 900", table, table);
+
+            assertUpdate(format("CREATE MATERIALIZED VIEW %s WITH (partitioned_by = ARRAY['nationkey']) " +
+                    "AS %s", view, baseQuery));
+
+            assertUpdate(format("REFRESH MATERIALIZED VIEW %s WHERE nationkey < 10", view, baseQuery), 380);
+
+            String viewQuery = format("SELECT name, custkey, nationkey from %s ORDER BY name", view);
+            baseQuery = format("%s ORDER BY name", baseQuery);
+
+            MaterializedResult viewTable = computeActual(viewQuery);
+            MaterializedResult baseTable = computeActual(baseQuery);
+            assertEquals(viewTable, baseTable);
```

Comment:
Lets add the plan comparison as well.

Reply:
Not sure if there's an easy way, the plan looks pretty complicated with exchanges happening for except.